tests/test.py

```diff
@@ -15,73 +15,84 @@
-        self.assertGreater(score, 0.8, "Failed with score = {0:.5f}".format(score))
+        score = clf.score(self.X_test, self.y_test)
+        self.assertGreaterEqual(score, 0.9, "Failed with score = {0:.5f}".format(score))
 
     def test_bin_classifier(self):
         clf = self.classifier_class(**self.kwargs)
-        bin_target = (self.iris.target == 2).astype(int)
-        clf.fit(self.iris.data, bin_target)
+        bin_target_train = (self.y_train == 2).astype(int)
+        bin_target_test = (self.y_test == 2).astype(int)
+        clf.fit(self.X_train, bin_target_train)
 
-        proba_sum = clf.predict_proba(self.iris.data).sum(axis=1)
-        np.testing.assert_almost_equal(proba_sum, np.ones(bin_target.shape[0]))
+        proba_sum = clf.predict_proba(self.X_test).sum(axis=1)
+        np.testing.assert_almost_equal(proba_sum, np.ones(bin_target_test.shape[0]))
 
-        score = clf.score(self.iris.data, bin_target)
-        print('Score: {0:.5f}'.format(score))
-        self.assertGreater(score, 0.8, "Failed with score = {0:.5f}".format(score))
+        score = clf.score(self.X_test, bin_target_test)
+        self.assertGreaterEqual(score, 0.9, "Failed with score = {0:.5f}".format(score))
 
     def test_string_y(self):
         clf = self.classifier_class(**self.kwargs)
-        y_str = np.array(self.iris.target, dtype=str)
-        y_str[y_str == '0'] = 'Zero'
-        y_str[y_str == '1'] = 'One'
-        y_str[y_str == '2'] = 'Two'
 
-        clf.fit(self.iris.data, y_str)
-        y_pred = clf.predict(self.iris.data)
-        score = accuracy_score(y_str, y_pred)
-        self.assertGreater(score, 0.95, "Failed with score = {0:.5f}".format(score))
+        clf.fit(self.X_train, self.y_str_train)
+        y_pred = clf.predict(self.X_test)
+        score = accuracy_score(self.y_str_test, y_pred)
+        self.assertGreaterEqual(score, 0.9, "Failed with score = {0:.5f}".format(score))
 
     def test_bin_string_y(self):
         clf = self.classifier_class(**self.kwargs)
-        y_str = np.array(self.iris.target, dtype=str)
-        y_str[y_str == '0'] = 'Zero'
-        y_str[y_str == '1'] = 'One'
-        y_str[y_str == '2'] = 'Two'
 
-        bin_X = self.iris.data[self.iris.target != 0]
-        y_str = y_str[y_str != 'Zero']
+        bin_X_train = self.X_train[self.y_train != 0]
+        bin_X_test = self.X_test[self.y_test != 0]
+        y_str_train = self.y_str_train[self.y_str_train != 'Zero']
+        y_str_test = self.y_str_test[self.y_str_test != 'Zero']
 
-        clf.fit(bin_X, y_str)
-        y_pred = clf.predict(bin_X)
-        score = accuracy_score(y_str, y_pred)
-        self.assertGreater(score, 0.95, "Failed with score = {0:.5f}".format(score))
+        clf.fit(bin_X_train, y_str_train)
+        y_pred = clf.predict(bin_X_test)
+        score = accuracy_score(y_str_test, y_pred)
+        self.assertGreaterEqual(score, 0.75, "Failed with score = {0:.5f}".format(score))
```

Comment:
`0.75` is appropriate threshold?

Reply:
Yeah, on my computer `score` is exactly 0.75. That's why `assertGreaterEqual`, not just `assertGreater`.

Reply:
Thanks! That is exactly what I want to know :smile: